### DISCUSSION TITLE
Full example source code please

### DISCUSSION THREAD

**Discussion by thoffmann-fms** [rpearce/react-medium-image-zoom#386] [Q&A] [answered]
Page https://rpearce.github.io/react-medium-image-zoom/?path=/story/img--regular looks very useful since it uses separate buttons to do variable zooming, but I can't seem to reverse engineer exactly how this is done.  Could someone post the React source for this please?

**Comment by rpearce** [comment]
Hi, @thoffmann-fms! I'm the author  and maintainer of this open source, BSD-3-license library.

I'm not quite sure I understand your question, so can you help me so I can help you?

Are you asking for help on how to use this library in your code, or are you asking how to copy and paste the source code and use that?

In either case, I recommend including attribution via the LICENSE file in the project.

**Comment by thoffmann-fms** [comment]
Hello, thanks for responding.  I would like to see the React source specifically for the URL above.  It looks very useful, but I can't reverse engineer what the various props, events, listeners, etc. would be by reading the resulting JS via browser Dev Tools.

**Comment by rpearce** [comment]
@thoffmann-fms, are you interested in "reverse engineering" this for learning / academic purposes? If you're looking to copy the code without proper attribution, I believe that would be in violation of [`react-medium-image-zoom`'s LICENSE](https://github.com/rpearce/react-medium-image-zoom/blob/main/LICENSE).

If you are interested in simply browsing the source code, you can do so here:

* Project source: https://github.com/rpearce/react-medium-image-zoom/tree/main/source
* Example source: https://github.com/rpearce/react-medium-image-zoom/blob/16ebc24d38c28739f39c99b1f6bb616f7b0adb93/stories/Img.stories.tsx#L38-L52

Additionally, if you are merely curious about the techniques used here, I can try to answer any questions you may have.

**Reply by thoffmann-fms** [reply]
I'm looking only for the details on the samples of how to use it, specifically the buttons that zoom in/out a few % at a time. The snippet you provided above shows how to wrap an area, but I don't understand how to change the zoom in/out level in a stepped mode.  Also, I note that clicking those zoom buttons zoom more than just the <Zoom /> wrapped area but the entire <main /> block so somethings\ is pointing at more than just the image to zoom and in my case, I want only the image to change not any of the text above/below it to be affected.

**Reply by rpearce** [reply, marked as answer]
If you're using version `5.1.2`, you should only wrap the `<Zoom>` around a single image element.

The [Basic Usage section in the README](https://github.com/rpearce/react-medium-image-zoom#basic-usage) should cover exactly how to use this component. Can you give that a read and see if that makes sense and solves your problem?
